lib/lighthouse/letters_generator/service.rb

```diff
@@ -33,6 +33,70 @@ class Service < Common::Client::Base
 
```

Comment:
The changes here look bigger than they are. I added the new get letter method, and the rest of the changes are just moving methods into a private section and reducing redundancy. Comments in the current code indicated that we would like to reduce that redundancy, so I did it.